src/frontend/sharedComponents/Editor/index.tsx

```diff
@@ -131,7 +134,45 @@ export const Editor = ({
                     <React.Suspense
                       key={att.driveDiscoveryId + att.hash + att.type}
                       fallback={<ThumbnailLoader size={size} />}>
-                      <SavedPhotoThumbnailImage size={size} photo={att} />
+                      <SavedPhotoThumbnailImage
+                        size={size}
+                        photo={att}
+                        onPress={() => {
+                          if (!observationId) {
+                            navigate('PhotoPreviewModal', {
+                              photo: att,
+                            });
+                            return;
+                          }
+
+                          // TODO: Ideally use the core-react hooks but the error handling makes it tricky
+                          // since a boundary would need to be set up somewhere. Probably requires some bigger changes
+                          // to how this Editor component is implemented.
+                          projectApi.observation
+                            .getByDocId(observationId)
+                            .then(observation => {
+                              return projectApi.$originalVersionIdToDeviceId(
+                                observation.originalVersionId,
+                              );
+                            })
+                            .then(createdByDeviceId => {
+                              navigate('PhotoPreviewModal', {
+                                photo: att,
+                                observationDocId: observationId,
+                                createdByDeviceId,
+                                validatedByCoMapeo: true,
+                              });
+                            })
+                            .catch(() => {
+                              // TODO: Does it make sense to navigate to the screen if we fail to retrieve `createdByDeviceId`?
+                              navigate('PhotoPreviewModal', {
+                                photo: att,
+                                observationDocId: observationId,
+                                validatedByCoMapeo: true,
+                              });
```

Comment:
Highlighting this. not sure how often this would happen and I figured navigating to the screen with some missing info is better than not doing so at all.

separately, should this log to sentry if this happens?